### PR TITLE
Fixed "answer provided by user" issue on reCAPTCHA

### PR DESCRIPTION
**Pull request by rifatwahid**
Pull Request for Issue # .
"answer provided by user" issue on reCAPTCHA, Specially it's conflicting if multiple reCAPTCHA found on the respective page.

### Summary of Changes
Worked on "onCheckAnswer" function. The "$code" parameter's value wasn't implementing inside the function. While you call "onCheckAnswer" function, it used to take "response" from the form which was conflicting if multiple reCAPTCHA found on the respective page.

### Testing Instructions
Create multiple contact forms in one page with reCAPTCHA (it should be ajax contact form, We tested by SP Page Builder) and submit the second or third form with complete reCAPTCHA.

### Expected result
if anyone sends any value to "$code" parameter on "onCheckAnswer" function, it should take the "$code" parameter's value "$code" as "response".

### Actual result
Worked on "onCheckAnswer" function. The "$code" parameter's value wasn't implementing inside the function. While you call "onCheckAnswer" function, it used to take "response" from the form which was conflicting if multiple reCAPTCHA found on the respective page.

### Documentation Changes Required
No

### CODE REVIEW

File: plugins/captcha/recaptcha/recaptcha.php [hunk 1]
```diff
@@ -146,13 +146,13 @@ public function onCheckAnswer($code = null)
 		{
 			case '1.0':
 				$challenge = $input->get('recaptcha_challenge_field', '', 'string');
-				$response  = $input->get('recaptcha_response_field', '', 'string');
+				$response  = ($code) ? $code : $input->get('recaptcha_response_field', '', 'string');
```
Comment (SharkyKZ) [inline plugins/captcha/recaptcha/recaptcha.php]:
```suggestion
				$response  = $code ? $code : $input->get('recaptcha_response_field', '', 'string');
```

File: plugins/captcha/recaptcha/recaptcha.php [hunk 2]
```diff
@@ -146,13 +146,13 @@ public function onCheckAnswer($code = null)
...
 				$spam      = ($challenge === '' || $response === '');
 				break;
 			case '2.0':
 				// Challenge Not needed in 2.0 but needed for getResponse call
 				$challenge = null;
-				$response  = $input->get('g-recaptcha-response', '', 'string');
+				$response  = ($code) ? $code : $input->get('g-recaptcha-response', '', 'string');
```
Comment (SharkyKZ) [inline plugins/captcha/recaptcha/recaptcha.php]:
```suggestion
				$response  = $code ? $code : $input->get('g-recaptcha-response', '', 'string');
```

**Comment by Quy** [comment]
@HLeithner Please decide if this PR for 3.9 or #26854 for 3.10.

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on f7f7291d1db026f6e4bf060ef438787d90034934<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26882">issues.joomla.org/tracker/joomla-cms/26882</a>.</sub>

File: plugins/captcha/recaptcha/recaptcha.php [hunk 3]
```diff
@@ -146,13 +146,13 @@ public function onCheckAnswer($code = null)
 		{
 			case '1.0':
 				$challenge = $input->get('recaptcha_challenge_field', '', 'string');
-				$response  = $input->get('recaptcha_response_field', '', 'string');
+				$response  = $code ? $code : $input->get('recaptcha_response_field', '', 'string');
```
Comment (SharkyKZ) [inline plugins/captcha/recaptcha/recaptcha.php:149]:
```suggestion
				$response  = $code ?: $input->get('recaptcha_response_field', '', 'string');
```

File: plugins/captcha/recaptcha/recaptcha.php [hunk 4]
```diff
@@ -146,13 +146,13 @@ public function onCheckAnswer($code = null)
...
 				$spam      = ($challenge === '' || $response === '');
 				break;
 			case '2.0':
 				// Challenge Not needed in 2.0 but needed for getResponse call
 				$challenge = null;
-				$response  = $input->get('g-recaptcha-response', '', 'string');
+				$response  = $code ? $code : $input->get('g-recaptcha-response', '', 'string');
```
Comment (SharkyKZ) [inline plugins/captcha/recaptcha/recaptcha.php:155]:
```suggestion
				$response  = $code ?: $input->get('g-recaptcha-response', '', 'string');
```

**Comment by SharkyKZ** [comment]
Are there any free extensions to test this with? Contact forms are only in Pro version of SP Page Builder.

**Comment by rifatwahid** [comment]
Thanks for the test. I can provide you the SP Page Builder pro privately for test with it.

**Comment by SharkyKZ** [comment]
OK, send it to sharkykz@gmail.com please. And could you implement code simplifications suggested above. Thanks.

**Comment by mamunhpath** [comment]
> OK, send it to [sharkykz@gmail.com](mailto:sharkykz@gmail.com) please. And could you implement code simplifications suggested above. Thanks.

I have send you a SP Page Builder copy. Thanks.

**Comment by mamunhpath** [comment]
@SharkyKZ From Page Builder addon settings you need to change default captcha to Google reCaptcha.

**Comment by SharkyKZ** [comment]
Thanks. PR works fine. Just please update it according to code suggestions above.

**Comment by rifatwahid** [comment]
is everything ok now, please? Please note, the same issue exists in the invisible reCaptcha also. will I request a pull again? or you can add the code?

**Comment by HLeithner** [comment]
Thanks for fixing this bug.

**Comment by rifatwahid** [comment]
Hi HLeithner,
You are always welcome, The same issue exists in the invisible reCaptcha aswell, Will I request a pull again as like this again? or, you guys will fix it?

- Thanks

**Comment by HLeithner** [comment]
@rifatwahid please provide a pull request. Because "you guys" means also you ;-)

**Comment by GCLW** [comment]
This broke a custom form module I use on my sites.
It returns a joomla error page with the error "invalid-input-response".

**Pull request merged**